cmd/process-agent/collector.go

```diff
@@ -285,6 +287,9 @@ func (l *Collector) postToAPI(endpoint config.APIEndpoint, checkPath string, bod
 	req.Header.Add("X-Dd-APIKey", endpoint.APIKey)
 	req.Header.Add("X-Dd-Hostname", l.cfg.HostName)
 	req.Header.Add("X-Dd-Processagentversion", Version)
+	if hasContainers {
```

Comment:
let's add 

```
else {
 req.Header.Add("X-Dd-HasContainer", "false")
}
```

this way we will be able to tell the difference between an message missing the header because the agent is too old, or because there are actually no containers.